zookeeper-server/src/main/java/org/apache/zookeeper/server/DataNode.java

```diff
@@ -37,6 +37,14 @@
  * 
  */
 public class DataNode implements Record {
+
+    // the digest value of this node, calculated from path, data and stat
+    private long digest;
```

Comment:
It's a single thread read/write this value for now, so it doesn't matter, but I agree it would be better to have it as a volatile in case we need to visit this in different thread in the future, will change that.